contracts/ProtocolReserve/RiskFundV2.sol

```diff
@@ -200,37 +200,22 @@ contract RiskFundV2 is AccessControlledV8, RiskFundV2Storage, IRiskFund {
             }
             uint256 distributedShare;
             for (uint256 i; i < poolsLength; ) {
-                uint256 poolShare = (poolAssetsFunds[pools[i]][tokenAddress] * EXP_SCALE) / assetReserves;
-                if (poolShare != 0) {
+                if (poolAssetsFunds[pools[i]][tokenAddress] != 0) {
+                    uint256 poolAmountShare;
                     if (i < (poolsLength - 1)) {
-                        distributedShare += updatePoolAssetsReserve(pools[i], tokenAddress, amountDiff, poolShare);
+                        poolAmountShare = (poolAssetsFunds[pools[i]][tokenAddress] * amount) / assetReserves;
+                        poolAssetsFunds[pools[i]][tokenAddress] -= poolAmountShare;
+                        distributedShare += poolAmountShare;
                     } else {
-                        uint256 distributedDiff = amountDiff - distributedShare;
-                        poolAssetsFunds[pools[i]][tokenAddress] -= distributedDiff;
-                        emit PoolAssetsDecreased(pools[i], tokenAddress, distributedDiff);
+                        poolAmountShare = amountDiff - distributedShare;
+                        poolAssetsFunds[pools[i]][tokenAddress] -= poolAmountShare;
```

Comment:
This statement is duplicated in the `if` block. We could move it out of the `else` block, and remove it from the `if` and `else` blocks

Reply:
Refactored, https://github.com/VenusProtocol/protocol-reserve/pull/29/commits/3e600898ea8522c50f1b1361e378866889de9f41